### ISSUE TITLE
Dropdown component fails to call event.preventDefault() and event.stopPropagation() when handling keyboard interactions

### ISSUE THREAD

**Issue by brichwin** [indiana-university/rivet-source#598]
## Describe the bug
Keyboard events that are part of the IU Rivet dropdown component's interactivity are continuing to propagate and be handled by the browser as usual despite being handled by Rivet. This is causing undesirable effects. For example, using the up arrow and down arrow keys not only interacts with the dropdown component (opens menu, moves between menu items), but also errantly scrolls the page.

The undesirable effects (such as page scrolling) reduce IU Rivet's credibility and also could cause issues for users of screen-magnification as page scrolling could take the menu completely out of their zoomed in view.  

## Expected behavior
Keystroke events handled by the dropdown component should not cause the default browser reaction and probably should not continue bubbling up the event chain. The IU Rivet js code should at a minimum call `event.preventDefault()` for each keystroke event handled by a component. It's a good idea to also call `event.stopPropagation()`.

## Relevant context (required)
- For an example of code that does not have this issue, note that in the [menu-button-actions.js code](https://w3c.github.io/aria-practices/examples/menu-button/js/menu-button-actions.js) on the w3c.github.io/aria-practices site, the keyboard handlers set a flag when they have handled a keystroke and right before a keyboard event handler ends, it checks the flag and then both stops the user agent's default response and stops the event from propagating to further event handlers:
```js
    if (flag) {
      event.stopPropagation();
      event.preventDefault();
    }
```
- Here is a Kaltura [video of the errant behavior of Rivet dropdown components due to failing to stop handled keyboard browser's default keyboard handling of keyboard events handled by the dropdown component](https://iu.mediaspace.kaltura.com/media/t/1_9rsw226y)

## Steps to reproduce

1. Navigate to [Rivet 1.8.2 Dropdown example](https://rivet.iu.edu/components/navigation/dropdown/#dropdown-example)
2. Set the browser window to a height of around 2 inches (so page scrolling becomes obvious)
3. Move keyboard focus to the "Navigation menu" dropdown button
4. Press the down arrow key to open the dropdown's menu. Note that the menu opens **and the page also scrolls**.
5. Use the down arrow key to move focus to one of the menu items
6. Use the up/down arrow keys to move between menu items. **Note that besides just moving between menu items, the page also scrolls up or down with each press of the up or down arrow key**.

**Comment by brichwin** [comment]
This issue is present in Rivet 1 and Rivet 2. Since the issue was discovered by a Canvas user, please address the Rivet 1 codebase as well.

**Comment by brichwin** [comment]
This issue also manifests in the tabs component: https://github.com/indiana-university/rivet-source/issues/602

**Comment by levimcg** [comment]
@brichwin Thanks again for reporting this. We'll definitely work on implementing the `preventDefault()` fix for arrow navigation.

I'll need to do some digging through old issues, but I remember specifically removing `event.stopPropigation()` from earlier implementations. There are times when the rivet dropdown/menu button needs to be used inside of other UI frameworks/libraries that implement a virtual DOM (e.g. React, Vue, etc.). When we call `event.stopPropigation()` in the case where a user clicks or arrows, etc. it can cause collisions with the synthetic events fired by the library's virtual DOM implementation, which breaks the functionality of the dropdown component.

In Rivet 1 specifically, we use an event delegation pattern where all (or most) event listeners are attached to the document instead of the specific DOM elements. We need the events to bubble up to the document so that we can implement handlers there, in one place rather than on a per element basis. We do this so that additional elements like dropdown, modals, tabs, etc. can be added to the DOM after page load without any additional initialization logic for event handlers.

With the event delegation model in place we don't need to call `event.stopPropigation()` because all of the widgets work this way and we implement logic to figure out which element/instance of the widget the event came from in the single event listener that is attached on `document.body`.

**Comment by levimcg** [comment]
This issue was addressed in #627 and is available in the [rivet-uits@1.8.3 release](https://github.com/indiana-university/rivet-source/releases/tag/v1.8.3)
